### ISSUE TITLE
Tags for resources erroring using pulumi_aws_native

### ISSUE THREAD

**Issue by OdmBkv** [pulumi/pulumi-aws-native#878]
### What happened?

The tagging functionality format does not accept the same format I use in the classic api.
for example using
```
aws_tags = {
    'function': cfg.get('function'),
    'application': cfg.get('application')}
```
has worked in the aws classic but this aws_native keeps erroring
```
ds_s3_resource_bucket = aws_native.s3.Bucket(resource_name= ds_well_comm_base_name
                                            , bucket_name= ds_well_comm_base_name
                                            , access_control = 'Private'
                                            , tags= aws_tags)
```
keeps erroring
```
    error: aws-native:s3:Bucket resource 'ds-well-comm-base': property tags[0].Key value {<nil>} has a problem: unknown property tags[0].Key
    error: aws-native:s3:Bucket resource 'ds-well-comm-base': property tags[0].Value value {<nil>} has a problem: unknown property tags[0].Value
```
I have tried other formats like a list of jsons. tags=[json.dumps({ 'function': cfg.get('function')}) but get errors like
```
...problem: tags[0] must be an object
or
...problem: tags must be an array
```

Also explict attempt to use the boto2.client('s3). put_bucket_tagging(...syntax....)
```
 s3_client.put_bucket_tagging(
    Bucket='ds-well-comm-base'
    , Tagging={
        'TagSet': [
            {
                'Key': 'string',
                'Value': 'string'
            },
        ]
    }
)
```

Like this 
```
ds_s3_resource_bucket = aws_native.s3.Bucket(resource_name= ds_well_comm_base_name
                                            , bucket_name= ds_well_comm_base_name
                                            , access_control = 'Private'
                                            , tags= [
                                                    {
                                                        'Key': 'string',
                                                        'Value': 'string'
                                                    },
                                                ]
                                            )
```

Gives this error
```
  aws-native:s3:Bucket (ds-well-comm-base-dev):
    error: aws-native:s3:Bucket resource 'ds-well-comm-base-dev': property tags[0].Key value {<nil>} has a problem: unknown property tags[0].Key
    error: aws-native:s3:Bucket resource 'ds-well-comm-base-dev': property tags[0].Value value {<nil>} has a problem: unknown property tags[0].Value
```

### Expected Behavior

Tags assigned to resource.

## 1st example using
```
aws_tags = {
    'function': cfg.get('function'),
    'application': cfg.get('application')}
```
has worked in the aws classic but this aws_native keeps erroring
```
ds_s3_resource_bucket = aws_native.s3.Bucket(resource_name= ds_well_comm_base_name
                                            , bucket_name= ds_well_comm_base_name
                                            , access_control = 'Private'
                                            , tags= aws_tags)
```
errors
```
    error: aws-native:s3:Bucket resource 'ds-well-comm-base': property tags[0].Key value {<nil>} has a problem: unknown property tags[0].Key
    error: aws-native:s3:Bucket resource 'ds-well-comm-base': property tags[0].Value value {<nil>} has a problem: unknown property tags[0].Value
```

## Attempt other format like this 
```
ds_s3_resource_bucket = aws_native.s3.Bucket(resource_name= ds_well_comm_base_name
                                            , bucket_name= ds_well_comm_base_name
                                            , access_control = 'Private'
                                            , tags= [
                                                    {
                                                        'Key': 'string',
                                                        'Value': 'string'
                                                    },
                                                ]
                                            )
```

Gives this error
```
  aws-native:s3:Bucket (ds-well-comm-base-dev):
    error: aws-native:s3:Bucket resource 'ds-well-comm-base-dev': property tags[0].Key value {<nil>} has a problem: unknown property tags[0].Key
    error: aws-native:s3:Bucket resource 'ds-well-comm-base-dev': property tags[0].Value value {<nil>} has a problem: unknown property tags[0].Value
```

## well_comm_infra > pulumi about
CLI
Version      3.62.0
Go Version   go1.20.2
Go Compiler  gc

Plugins
NAME        VERSION
aws         5.35.0
aws-native  0.57.0
awsx        1.0.2
docker      3.6.1
python      unknown

Host
OS       Microsoft Windows 10 Pro
Version  10.0.19044 Build 19044
Arch     x86_64

This project is written in python: executable='C:\GitHub\well_comm_infra\py_venv\python.exe' version='3.9.16
'

Current Stack: bkvcorp/well_comm_infra/dev

Found no resources associated with dev

Found no pending operations associated with dev

Backend
Name           pulumi.com
URL            https://app.pulumi.com/cliffclausen
User           cliffclausen
Organizations  cliffclausen, bkvcorp

Dependencies:
NAME                           VERSION
awswrangler                    2.20.1
backports.functools-lru-cache  1.6.4
brotlipy                       0.7.0
entrypoints                    0.4.0
flit_core                      3.8.0
importlib-resources            5.12.0
jupyterlab                     3.6.3
pip                            23.0.1
pkgutil_resolve_name           1.3.10
pulumi-aws-native              0.57.0
pulumi-awsx                    1.0.2
pylint                         2.17.2
pyOpenSSL                      23.1.1
PySocks                        1.7.1
wheel                          0.40.0
win-inet-pton                  1.1.0

### Additional context

Discussed in [Slack thread](https://app.slack.com/client/T85C88L30/CRH5ENVDX/thread/CRH5ENVDX-1681412064.572979)

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by OdmBkv** [comment]
@mikhailshilkov 
I can make that work, thank you, getting the syntax correct was killing me. 
For anyone that reads, taking classic dictionary used before, I put it in a list comprehension  for aws_native tags
`aws_tags_native = [aws_native.s3.BucketTagArgs(key=akey, value=avalue) for akey, avalue in aws_tags.items()]` 
then tags= aws_tags_native

**Comment by mikhailshilkov** [comment]
@OdmBkv Thank you for confirming that it works for you, and for the conversion snippet.

**Comment by cameronmcdougle** [comment]
I spent a considerable amount of time on this today and this thread saved my bacon.

A few of the errors I got in case it helps the searches:

`error: aws-native:iam:Role resource 'BotName_Role': property tags value {map[Client:{test}]} has a problem: tags must be an array`

`aws-native:iam:Role resource 'BotName_Role': property tags[0].Client value {<nil>} has a problem: unknown property tags[0].Client`

and snippets of code that worked:

```
aws_native.iam.Role(
    ...
    tags=[aws_native.iam.RoleTagArgs(key=key, value=value) for key, value in tags.items()],
)
```

I'm really not a fan of this implementation, and it's not documented with examples so far as I can see, nor was it immediately clear reading through the source code. 

Additionally, the fact that the TagsArgs is different for every resource (and I didn't get using its source from `_inputs` working calling it directly), is this worth raising an issue/feature request to abstract this? I'd prefer not to repeat that code all over.

**Comment by mjeffryes** [comment]
>Additionally, the fact that the TagsArgs is different for every resource (and I didn't get using its source from _inputs working calling it directly), is this worth raising an issue/feature request to abstract this? I'd prefer not to repeat that code all over.

Good news @cameronmcdougle, we recently fixed this in https://github.com/pulumi/pulumi-aws-native/pull/1348, you'll be able to use a shared `Tag` type (and even set  default tags) in the next release.

**Comment by cameronmcdougle** [comment]
> > Additionally, the fact that the TagsArgs is different for every resource (and I didn't get using its source from _inputs working calling it directly), is this worth raising an issue/feature request to abstract this? I'd prefer not to repeat that code all over.
> 
> Good news @cameronmcdougle, we recently fixed this in #1348, you'll be able to use a shared `Tag` type (and even set default tags) in the next release.

Oooooh excellent! Nice work 😎
